kubernetes-itests/src/test/java/io/fabric8/kubernetes/ServiceAccountIT.java

```diff
@@ -85,23 +91,35 @@ public void list() {
 
   @Test
   public void update() {
+    Ready<ServiceAccount> serviceAccountReady = new Ready<ServiceAccount>(ServiceAccount.class, client, "serviceaccount1", currentNamespace);
     serviceAccount1 = client.serviceAccounts().inNamespace(currentNamespace).withName("serviceaccount1").edit()
       .addNewSecret().withName("default-token-uudp").endSecret()
       .addNewImagePullSecret().withName("myregistrykey").endImagePullSecret()
       .done();
+    await().atMost(30, TimeUnit.SECONDS).until(serviceAccountReady);
     assertThat(serviceAccount1).isNotNull();
   }
 
   @Test
   public void delete() {
+    Ready<ServiceAccount> serviceAccountReady = new Ready<ServiceAccount>(ServiceAccount.class, client, "serviceaccount1", currentNamespace);
+    await().atMost(30, TimeUnit.SECONDS).until(serviceAccountReady);
     boolean bDeleted = client.serviceAccounts().inNamespace(currentNamespace).withName("serviceaccount1").delete();
     assertTrue(bDeleted);
   }
 
   @After
   public void cleanup() throws InterruptedException {
-    client.serviceAccounts().inNamespace(currentNamespace).delete();
-    // Wait for resources to get destroyed
-    Thread.sleep(30000);
+    try {
+      client.serviceAccounts().inNamespace(currentNamespace).delete();
+      // Wait for resources to get destroyed
+      Delete<ServiceAccount> serviceAccount1Delete = new Delete<ServiceAccount>(ServiceAccount.class, client, "serviceaccount1", currentNamespace);
+      Delete<ServiceAccount> serviceAccount2Delete = new Delete<ServiceAccount>(ServiceAccount.class, client, "serviceaccount2", currentNamespace);
+
+      await().atMost(30, TimeUnit.SECONDS).until(serviceAccount1Delete);
+      await().atMost(30, TimeUnit.SECONDS).until(serviceAccount2Delete);
```

Comment:
Can this be combined to one `&&` condition ?

Reply:
As in this ?
```await().atMost(30, TimeUnit.SECONDS).until(serviceAccount1Delete && serviceAccount2Delete);```

This doesn't work. data types of serviceAccountDelete instances aren't compatible for this boolean specific operation.